Andrii_Voinenko/Lesson9/task2.py

```diff
@@ -0,0 +1,10 @@
+def range_generator():
+    counter = 0
+    while counter <= 1000000000:
+        yield counter
+        counter += 1
+
+
+for number in range_generator():
+    if number % 2 == 0:
+        print(pow(number, 2))
```

Comment:
you can handle this inside decorator. it is better than calculating it in for each loop